### PR TITLE
feat: backend v2 refactoring

### PR DESCRIPTION
**Pull request by birdringxD**

### CODE REVIEW

File: backend/docker-compose.yaml [hunk 1]
```diff
@@ -1,55 +1,32 @@
...
-    image: semitechnologies/weaviate:1.20.3
-    container_name: weaviate
-    restart: unless-stopped
-    ports:
-      - "8091:8080"
-    env_file:
-      - .env
+  vec_db:
+    image: pgvector/pgvector:pg16
+    container_name: vec_db
```
Comment (HenryQW) [inline backend/docker-compose.yaml:5]: missing restart policy?

File: backend/docker-compose.yaml [hunk 2]
```diff
@@ -1,55 +1,32 @@
...
-    environment:
-      - WEAVIATE_URL=http://weaviate:8080
-      - WEAVIATE_API_KEYS=${WEAVIATE_API_KEYS}
+
+
 volumes:
-  weaviate_data:
+  pgvector_data:
   pg_data:
   redis_data:
```
Comment (HenryQW) [inline backend/docker-compose.yaml:32]: is redis no longer needed?

File: backend/openagent/index/feed_scrape.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return _fetch_feeds()
+    except Exception as e:
+        logger.error(f"Failed to fetch feeds from {platform}: {e}")
+        return {}
+
+
+if __name__ == "__main__":
+    curr_ts = int(datetime.datetime.now().timestamp())
+    _cursor = None
+    feeds = fetch_mirror_feeds(0, curr_ts, cursor=_cursor)
```
Comment (HenryQW) [inline backend/openagent/index/feed_scrape.py:61]: since_timestamp = 0
this is generating unnecessary requests to the server, consider a recent (6m to 1 year?) timestamp instead.

File: backend/openagent/index/pgvector_store.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        connection=settings.VEC_DB_CONNECTION,
+        use_jsonb=True,
+    )
+
+
+store = build_vector_store()
+
+
+if __name__ == "__main__":
+    docs = [
```
Comment (HenryQW) [inline backend/openagent/index/pgvector_store.py:26]: what are these docs for?

File: backend/openagent/index/feed_indexing.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            break
+        cursor = resp["meta"]["cursor"]
+        logger.info(f"fetched {len(resp['data'])} records, next cursor: {cursor}")
+
+        # get all the records
+        records = resp.get("data", [])
+        if len(records) == 0:
+            break
+
+        save_records(records)
```
Comment (HenryQW) [inline backend/openagent/index/feed_indexing.py:68]: should not have duplicated logic

File: backend/openagent/index/pgvector_store.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+from langchain.embeddings import OpenAIEmbeddings
+from langchain_core.documents import Document
+from langchain_postgres.vectorstores import PGVector
+
+from openagent.conf.env import settings
+
+load_dotenv()
+
+
+def build_vector_store() -> PGVector:
```
Comment (HenryQW) [inline backend/openagent/index/pgvector_store.py:11]: can we add generic support for other vector databases?

**Review by HenryQW — CHANGES_REQUESTED** [review]

File: backend/docker-compose.yaml (hunk 1 above)
Comment (naaive) [reply backend/docker-compose.yaml:5]: added

File: backend/docker-compose.yaml (hunk 2 above)
Comment (naaive) [reply backend/docker-compose.yaml]: yes

File: backend/openagent/index/pgvector_store.py (hunk 6 above)
Comment (naaive) [reply backend/openagent/index/pgvector_store.py:10]: any subclass implement ` Vector Store(langchain_core/vectorstores.py)` should be ok

File: backend/openagent/index/pgvector_store.py (hunk 4 above)
Comment (naaive) [reply backend/openagent/index/pgvector_store.py]: useless code, deleted

**Comment by HenryQW** [comment]
Hey @birdringxD, while reviewing your PR, I'd suggest the following code changes:

👉 [**Code Suggestion for #11**](https://gitkraken.dev/drafts/0aea999d-e2da-4d80-bca4-d9d54c553140?source=PRComment)
>https://github.com/RSS3-Network/OpenAgent/pull/11

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/0aea999d-e2da-4d80-bca4-d9d54c553140?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPTGMzRkZjNXZyZ3V1Il0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

**Review by HenryQW — APPROVED** [review]

**Pull request merged**